src/emulator/functionsEmulator.ts

```diff
@@ -514,6 +517,79 @@ You can probably fix this by running "npm install ${
     return loadTriggers();
   }
 
+  addRealtimeDatabaseTrigger(
+    projectId: string,
+    definition: EmulatedTriggerDefinition
+  ): Promise<any> {
+    const databasePort = EmulatorRegistry.getPort(Emulators.DATABASE);
+    if (!databasePort) {
+      EmulatorLogger.log(
+        "INFO",
+        `Ignoring trigger "${
+          definition.name
+        }" because the Realtime Database emulator is not running.`
+      );
+      return Promise.resolve();
+    }
+    if (definition.eventTrigger === undefined) {
+      EmulatorLogger.log(
+        "WARN",
+        `Event trigger "${definition.name}" has undefined "eventTrigger" member`
+      );
+      return Promise.reject();
+    }
+    /*
+     * The Realtime Database emulator expects the `path` field in its trigger
+     * definition to be relative to the database root.
+     */
+    const pathPattern = new RegExp("^projects/[^/]*/instances/[^/]*/refs(/.*)$");
+    const result: string[] | null = pathPattern.exec(definition.eventTrigger.resource);
```

Comment:
Do we need explicit type definition here? What happens if you just let ts infer?

Reply:
We don't need an explicit definition but I think it's nice to have one here since it expresses what happens in the success and failure cases. If you want to get rid of it that's fine with me.